src/App.tsx

```diff
@@ -1,26 +1,237 @@
+      setError('Title should not be empty');
+      setTimeout(() => setError(''), 3000);
+
+      return;
+    }
+
+    if (inputRef.current) {
+      inputRef.current.disabled = true;
+    }
+
+    setTempTodo({
+      id: 0,
+      title: newTodoTitle,
+      userId: USER_ID,
+      completed: false,
+    });
+
+    setLoadingTodos(curr => [...curr, 0]);
+
+    postTodo({ title: todoTitle, userId: USER_ID, completed: false })
+      .then(newTodo => {
+        setTodos(curTodos => [...curTodos, newTodo]);
+
+        setQuery('');
+      })
+      .catch(() => {
+        setError('Unable to add a todo');
+        setTimeout(() => setError(''), 3000);
+      })
+      .finally(() => {
+        if (inputRef.current) {
+          inputRef.current.disabled = false;
+          inputRef.current.focus();
+        }
+
+        setTempTodo(null);
+        setLoadingTodos(curr => curr.filter(todoId => todoId !== 0));
+      });
+  };
+
+  const handleDeleteTodo = (todoId: number) => {
+    setLoadingTodos(curr => [...curr, todoId]);
+
+    deleteTodo(todoId)
+      .then(() =>
+        setTodos(currTodos => currTodos.filter(todo => todo.id !== todoId)),
+      )
+      .catch(() => {
+        setError('Unable to delete a todo');
+        setTimeout(() => setError(''), 3000);
+      })
+      .finally(() =>
+        setLoadingTodos(curr =>
+          curr.filter(deletingTodoId => todoId !== deletingTodoId),
+        ),
+      );
+
+    inputRef.current?.focus();
+  };
+
+  const completedTodos = todos.filter(todo => todo.completed);
+
+  const deleteAllComleted = () => {
+    completedTodos.forEach(todo => handleDeleteTodo(todo.id));
+  };
+
+  const handleToggleStatus = (todo: Todo) => {
+    setLoadingTodos(curr => [...curr, todo.id]);
+
+    updateTodo({ ...todo, completed: !todo.completed })
+      .then(updatedTodo =>
+        setTodos(currTodos =>
+          currTodos.map(currTodo =>
+            currTodo.id === updatedTodo.id ? updatedTodo : currTodo,
+          ),
+        ),
+      )
+      .catch(() => setError('Unable to update a todo'))
+      .finally(() => {
+        setLoadingTodos(curr => curr.filter(id => id !== todo.id));
+      });
+  };
+
+  const handleAllToggleStatus = () => {
+    if (activeTodos.length > 0) {
+      activeTodos.forEach(handleToggleStatus);
+    } else {
+      comletedTodos.forEach(handleToggleStatus);
+    }
+  };
+
+  // const handleUpdateTodo = async(updatedTodo: Todo) => {
+  //   setLoadingTodos(curr => [...curr, updatedTodo.id])
+
+  //   updateTodo(updatedTodo)
+  //     .then(() =>
+  //     setTodos(
+  //       todos.map(todo => (todo.id === updatedTodo.id ? updatedTodo : todo)),
+  //       ),
+  //     )
+  //     .catch(() => {
+  //       setError('Unable to update a todo');
+  //       throw new Error();
+  //     }
+
+  //     )
+  //     .finally(() => {
+  //       setLoadingTodos(curr => curr.filter(id => id !== updatedTodo.id))
+  //     })
+  //     // throw new Error();
+  // }
```

Comment:
Consider removing all redundant comments

Reply:
Not fixed